charts/keycloak/templates/_helpers.tpl

```diff
@@ -125,12 +125,23 @@ Create the name for the database password secret key.
 */}}
 {{- define "keycloak.dbPasswordKey" -}}
 {{- if .Values.keycloak.persistence.existingSecret -}}
-  {{- .Values.keycloak.persistence.existingSecretKey -}}
+  {{- required "missing existingSecretPasswordKey" .Values.keycloak.persistence.existingSecretPasswordKey -}}
```

Comment:
Why do you use different logic for username and password key?

Reply:
No good reason. Those definitions now use the same logic.

Reply:
No good reason. Those definitions now use the same logic.

One can now be configured without the other (which will fall back to  a default). 
This is more flexible than forcing the user to set both or neither.